client-sdk/go/crypto/signature/context.go

```diff
@@ -1,27 +1,68 @@
 package signature
 
 import (
+	"encoding/hex"
+
 	"github.com/oasisprotocol/oasis-core/go/common"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
 )
 
 const chainContextSeparator = " for chain "
 
-// Context is the chain domain separation context.
-type Context string
+// SignContext stores runtime ID, consensus chain context and tx-specific details
+// as defined in ADR 14 appropriate for CBOR encoding.
+type SignContext struct {
+	base         []byte
+	RuntimeID    string `json:"runtime_id"`
+	ChainContext string `json:"chain_context"`
+	OrigTo       string `json:"orig_to,omitempty"`
+}
+
```

Comment:
This type should probably have a method that returns a copy with a changed base? As otherwise it is not possible to use a different base once constructed.

E.g. something like:
```golang
func (sc *SignContext) WithBase(base []byte) *SignContext {
    return &SignContext{
        RuntimeID: sc.RuntimeID,
        ChainContext: sc.ChainContext,
        OrigTo: sc.OrigTo,
        base: base,
    }
}
```